integtest.sh

```diff
@@ -102,17 +102,84 @@ echo -e "Test Files List:"
 echo $TEST_FILES | tr ',' '\n'
 echo "BROWSER_PATH: $BROWSER_PATH"
 
+# Array to store remote cypress workflow background processes IDs when run in parallel
+declare -a all_process_pids
+
+run_remote_cypress() {
+    local repo="$1"
+    local workflow_name="$2"
+    local os_url="$3"
+    local osd_url="$4"
+    local branch_ref="$5"
+
+    # Call the remoteCypress.sh script with the required arguments
+    source remoteCypress.sh -r "$repo" -w "$workflow_name" -o "$os_url" -d "$osd_url" -b "$branch_ref" &
+    bg_process_pid=$!
+    echo "PID for the repo $repo is : $bg_process_pid"
+    all_process_pids+=($bg_process_pid)
+}
+
+# Read inputs from the manifest file using jq
+REMOTE_MANIFEST_FILE="remote_cypress_manifest.json"
+
+# Parse the JSON file using jq and iterate over the components array
+components=$(jq -c '.components[]' "$REMOTE_MANIFEST_FILE")
+release_version=$(jq -r '.build.version' "$REMOTE_MANIFEST_FILE")
+echo "Components: $components"
+echo "Release version: $release_version"
+
+for component in $components; do
+    echo "Processing for the component: $component"
+
+    repo=$(echo "$component" | jq -r '.["repository"]')
+    workflow_name=$(echo "$component" | jq -r '.["workflow-name"]')
+    os_url=$(echo "$component" | jq -r '.["opensearch"]')
+    osd_url=$(echo "$component" | jq -r '.["opensearch-dashboards"]')
+    branch_ref=$(echo "$component" | jq -r '.["ref"]')
+
+    # Set default values if the opensearch and opensearch-dashboards are not set in the manifest
+    os_url=${os_url:-https://artifacts.opensearch.org/releases/bundle/opensearch/$release_version/opensearch-$release_version-linux-x64.tar.gz}
+    osd_url=${osd_url:-https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/$release_version/opensearch-dashboards-$release_version-linux-x64.tar.gz}
+
+    echo "repo: $repo"
+    echo "workflow_name: $workflow_name"
+    echo "os_url: $os_url"
+    echo "osd_url: $osd_url"
+    echo "branch_ref: $branch_ref"
+
+    # Call the function for each component
+    run_remote_cypress "$repo" "$workflow_name" "$os_url" "$osd_url" "$branch_ref" 
```

Comment:
i think we should add a flag or env variable that enables us to conditional run the run remote cypress tests are not. since this repo is now being locked as well to commits a new build will have to be made if there are any issues. so it pre-planning a feature flag could be a fallback to disabling this if it causes issues.

Reply:
I referenced the [developer guide doc](https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/DEVELOPER_GUIDE.md#experimental-features) about setting up flag for experimental features and its tests. As my feature is not actually running cypress tests but instead running orchestrator workflows, I've introduced a simple boolean flag as part of `integtest.sh` script params defaulting to `true`. Build script which triggers `integtest.sh` can set the feature flag to true/false as required. Is this the behavior you were expecting or is there some other way to configure this setting in FTRepo?